org.eclipse.jdt.core.manipulation/core extension/org/eclipse/jdt/internal/corext/fix/SwitchFixCore.java

```diff
@@ -237,6 +238,9 @@ private List<SwitchCaseSection> filterDuplicateCaseValues(final List<SwitchCaseS
 
 					for (Expression expression : sourceCase.literalExpressions) {
 						Object constantValue= expression.resolveConstantExpressionValue();
+						if(constantValue == null && expression.resolveTypeBinding().isEnum()) {
```

Comment:
You always have to check for null with binding resolution.  Sometimes it is just the fact that there is an error somewhere else in the file.